lib/actions.js

```diff
@@ -332,26 +332,27 @@ actions.ChangeSetNotExecutableError = error('ChangeSetNotExecutableError');
  * @param {function} callback - a function that will be provided with the changeset
  * details once the changeset is in CREATE_COMPLETE, DELETE_COMPLETE, or FAILED state
  */
-function describeChangeset(cfn, name, changesetId, nextToken, callback) {
-
-  let changesetDescriptions;
-  (function callAPI(cfn, name, changesetId, nextToken, callback) {
-    cfn.describeChangeSet({ ChangeSetName: changesetId, StackName: name, NextToken: nextToken }, function(err, data) {
-      if (err) return callback(err);
-
-      if (changesetDescriptions) {
-        let changes = [].concat(changesetDescriptions.Changes, data.Changes);
-        changesetDescriptions = data;
-        changesetDescriptions.Changes = changes;
-      } else {
-        changesetDescriptions = data;
-      }
-      if ((data.Status === 'CREATE_COMPLETE' || data.Status === 'FAILED' || data.status === 'DELETE_COMPLETE') && !data.NextToken)
-        return callback(null, changesetDescriptions);
-
-      setTimeout(callAPI, 1000, cfn, name, changesetId, data.NextToken, callback);
-    });
-  })(cfn, name, changesetId, nextToken, callback);
+function describeChangeset(cfn, name, changesetId, callback) {
+
+  var changesetDescriptions;
+
+  cfn.describeChangeSet({ ChangeSetName: changesetId, StackName: name })
+  .eachPage((err, data, done) => {
+    if (err) return callback(err);
+
+    if (changesetDescriptions) {
+      var changes = [].concat(changesetDescriptions.Changes, data.Changes);
+      changesetDescriptions = data;
+      changesetDescriptions.Changes = changes;
+    } else {
+      changesetDescriptions = data;
+    }
+
+    if ((data.Status === 'CREATE_COMPLETE' || data.Status === 'FAILED' || data.status === 'DELETE_COMPLETE'))
+      return callback(null, changesetDescriptions);
+
+    done();
```

Comment:
Something seems odd here. I believe the `.eachPage(callback)` function indicates that you've gotten all the pages by firing your callback function with `data === null`. It seems like you should `.eachPage()` until you get to that condition, and then after that you should `callback(null, changesetDescriptions)`. 

Does that make sense?